pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -674,6 +733,56 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         _require(tokenData != 0, Errors.INVALID_TOKEN);
     }
 
+    // Join/exit callbacks
+
+    function _beforeJoinExit(
+        uint256[] memory,
+        uint256[] memory,
+        uint256
+    ) internal virtual override {
+        _collectAumManagementFees();
+    }
+
+    /**
+     * @dev Calculates the AUM fees accrued since the last collection and pays it to the pool manager.
+     * This function is called automatically on joins and exits.
+     */
+    function _collectAumManagementFees() internal {
+        uint256 lastCollection = _lastAumFeeCollectionTimestamp;
+        uint256 currentTime = block.timestamp;
+
+        // Collect fees based on the time elapsed
+        if (currentTime > lastCollection) {
+            // Reset the collection timer to the current block
+            _lastAumFeeCollectionTimestamp = currentTime;
+
+            uint256 managementAumFeePercentage = getManagementAumFeePercentage();
+
+            // `lastCollection == 0` means that we're in the first attempt to collect AUM fees
+            // For gas reasons we only collect AUM fees from this point onwards so perform an early return if so.
```

Comment:
These are not 'gas reasons': we can't collect fees the first time because we don't know what the elapsed time is (as we don't know when the pool was created).

We should also note in comments that since this function is called after all joins/exits, lastCollection will be automatically initialized to the time of the first non-initialziation join.

Reply:
> These are not 'gas reasons': we can't collect fees the first time because we don't know what the elapsed time is (as we don't know when the pool was created).

Well tbf we don't know that because it costs us gas on each join in order to check if we are initialising the pool.